packages/aws-cdk/test/api/hotswap/hotswap-test-setup.ts

```diff
@@ -60,16 +96,22 @@ export function stackSummaryOf(logicalId: string, resourceType: string, physical
 export class HotswapMockSdkProvider {
   public readonly mockSdkProvider: MockSdkProvider;
 
-  constructor() {
+  constructor(rootStackName?: string, childStacks?: string[]) {
     this.mockSdkProvider = new MockSdkProvider({ realSdk: false });
 
     this.mockSdkProvider.stubCloudFormation({
       listStackResources: ({ StackName: stackName }) => {
-        if (stackName !== STACK_NAME) {
-          throw new Error(`Expected Stack name in listStackResources() call to be: '${STACK_NAME}', but received: ${stackName}'`);
+        if (rootStackName) {
+          if (stackName !== rootStackName) {
+            if (childStacks && !(childStacks.includes(stackName))) {
```

Comment:
Can we collapse this into one `if`? Right now, this reads really strangely.

```suggestion
          if (stackName !== rootStackName && !childStacks.includes(stackName)) {
```